src/pipecat/services/assemblyai/stt.py

```diff
@@ -73,97 +90,176 @@ async def cancel(self, frame: CancelFrame):
+
+        except Exception as e:
+            logger.error(f"Fatal error in receive handler: {e}")
+
+    def _parse_message(self, message: Dict[str, Any]) -> BaseMessage:
+        """Parse a raw message into the appropriate message type."""
+        msg_type = message.get("type")
+
+        if msg_type == "Begin":
+            return BeginMessage.parse_obj(message)
+        elif msg_type == "Turn":
+            return TurnMessage.parse_obj(message)
+        elif msg_type == "Termination":
+            return TerminationMessage.parse_obj(message)
+        else:
+            raise ValueError(f"Unknown message type: {msg_type}")
+
+    async def _handle_message(self, message: Dict[str, Any]):
+        """Handle AssemblyAI WebSocket messages."""
+        try:
+            parsed_message = self._parse_message(message)
+
+            if isinstance(parsed_message, BeginMessage):
+                logger.debug(
+                    f"Session Begin: {parsed_message.id} (expires at {parsed_message.expires_at})"
+                )
+            elif isinstance(parsed_message, TurnMessage):
+                await self._handle_transcription(parsed_message)
+            elif isinstance(parsed_message, TerminationMessage):
+                await self._handle_termination(parsed_message)
+        except Exception as e:
+            logger.error(f"Error handling message: {e}")
 
-        def on_close():
-            """Callback for when the connection to AssemblyAI is closed."""
-            logger.info(f"{self}: Disconnected from AssemblyAI")
-
-        self._transcriber = aai.RealtimeTranscriber(
-            sample_rate=self.sample_rate,
-            encoding=self._settings["encoding"],
-            on_data=on_data,
-            on_error=on_error,
-            on_open=on_open,
-            on_close=on_close,
+    async def _handle_termination(self, message: TerminationMessage):
+        """Handle termination message."""
+        self._received_termination = True
+        self._termination_event.set()
+
+        logger.info(
+            f"Session Terminated: Audio Duration={message.audio_duration_seconds}s, "
+            f"Session Duration={message.session_duration_seconds}s"
         )
-        self._transcriber.connect()
+        await self.push_frame(EndFrame())
 
-    async def _disconnect(self):
-        """Disconnect from the AssemblyAI service and clean up resources."""
-        if self._transcriber:
-            self._transcriber.close()
-            self._transcriber = None
+    async def _handle_transcription(self, message: TurnMessage):
+        """Handle transcription results."""
+        if not message.transcript:
+            return
+        await self.stop_ttfb_metrics()
+        if message.end_of_turn:
+            await self.push_frame(
+                TranscriptionFrame(
+                    message.transcript,
+                    "",  # participant
+                    time_now_iso8601(),
+                    self._language,
```

Comment:
I just merged a PR that adds an optoinal `result` field to `TranscritionFrame` and `InterimTranscriptionFrame`. The idea is to set this field to whatever raw response the STT service returned. In this case, I think we would do `result=message`. Then, the user can inspect what the STT returned.